jetty-client/src/main/java/org/eclipse/jetty/client/HttpRedirector.java

```diff
@@ -292,7 +302,8 @@ private Request redirect(Request request, Response response, Response.CompleteLi
         Integer redirects = (Integer)conversation.getAttribute(ATTRIBUTE);
         if (redirects == null)
             redirects = 0;
-        if (redirects < client.getMaxRedirects())
+        int maxRedirects = client.getMaxRedirects();
+        if (maxRedirects < 0 || redirects < maxRedirects)
```

Comment:
I would expect a setting of `maxRedirects = -1` to mean no redirects are being performed. (same as what `maxRedirects = 0` does now)
Not that there is an unlimited number of redirects.

Reply:
No, whether to follow redirects or not is controlled by a boolean `HttpClient.followRedirects`. If that is true, we need to limit the max number of redirects, so `-1` or `0` act as unlimited.